controller.go

```diff
@@ -240,17 +244,24 @@ func setupManager(
 	if err = invclean.SetupWithManager(mgr); err != nil {
 		return nil, err
 	}
+	failcount := controllers.NewFailureCounter()
+	successcount := controllers.NewSuccessCounter()
 	invmail := &controllers.InvitationEmailReconciler{
 		Client:         mgr.GetClient(),
 		Scheme:         mgr.GetScheme(),
 		Recorder:       mgr.GetEventRecorderFor("invitation-email-controller"),
 		BaseRetryDelay: invEmailBaseRetryDelay,
 		MailSender:     mailSender,
+		SuccessCounter: successcount,
```

Comment:
Personal preference: This could be encapsulated more more nicely by making those fields private and adding a `NewInvitationEmailReconciler` method.

Reply:
They'll need to be registered in a Metric Registry, though, so we'll at least need a getter.

Reply:
`metrics.Registry` seems to be a singleton accessible from everywhere. But yeah then it's the problem with potentially having multiple instances of this `Reconciler`. Fine as is i guess.